apps/lockfile-explorer-web/src/containers/LockfileEntryDetailsView/index.tsx

```diff
@@ -0,0 +1,200 @@
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { pushToStack, selectCurrentEntry } from '../../store/slices/entrySlice';
+import { ReactNull } from '../../types/ReactNull';
+import { LockfileEntry } from '../../parsing/LockfileEntry';
+
+enum InfluencerTypes {
+  DETERMINANT,
+  TRANSITIVE_REFERRER
+}
+
+interface IInfluencerType {
+  entry: LockfileEntry;
+  type: InfluencerTypes;
+}
+
+export const LockfileEntryDetailsView = (): JSX.Element | ReactNull => {
+  const selectedEntry = useAppSelector(selectCurrentEntry);
+  const dispatch = useAppDispatch();
+
+  const [inspectDep, setInspectDep] = useState<LockfileDependency | null>(null);
+  const [influencers, setInfluencers] = useState<IInfluencerType[]>([]);
+
+  useEffect(() => {
+    if (selectedEntry) {
+      setInspectDep(null);
+    }
+  }, [selectedEntry]);
+
+  const selectResolvedEntry = useCallback(
+    (dependency) => () => {
+      if (inspectDep && inspectDep.name === dependency.name) {
+        if (dependency.resolvedEntry) {
+          dispatch(pushToStack(dependency.resolvedEntry));
+        } else {
+          console.error('No resolved entry for dependency: ', dependency);
+        }
+      } else if (selectedEntry) {
+        setInspectDep(dependency);
+        // calculate influencers
+        const stack = [selectedEntry];
+        const determinants = new Set<LockfileEntry>();
+        const transitiveReferrers = new Set<LockfileEntry>();
+        const visitedNodes = new Set<LockfileEntry>();
+        visitedNodes.add(selectedEntry);
+        while (stack.length) {
+          const currEntry = stack.pop();
+          if (currEntry) {
+            for (const referrer of currEntry.referencers) {
+              let flag = false;
+              for (const dep of referrer.dependencies) {
+                if (dep.name === dependency.name) {
+                  determinants.add(referrer);
+                  flag = true;
+                  break;
+                }
+              }
+              if (!flag) {
+                if (referrer.transitivePeerDependencies.has(dependency.name)) {
+                  transitiveReferrers.add(referrer);
+                } else {
+                  console.error('Found referrer that does not correctly state tpd: ', referrer);
+                }
+                for (const referencer of currEntry.referencers) {
+                  if (!visitedNodes.has(referencer)) {
+                    stack.push(referencer);
+                    visitedNodes.add(referencer);
+                  }
+                }
+              }
+            }
+          }
+        }
+        const influencers: IInfluencerType[] = [];
+        for (const det of determinants.values()) {
+          influencers.push({
+            entry: det,
+            type: InfluencerTypes.DETERMINANT
+          });
+        }
+        for (const ref of transitiveReferrers.values()) {
+          influencers.push({
```

Comment:
Spell out `det`->`determinant`, `ref`->`referrer`, etc

Reply:
changed, I renamed referrer above to reference, should these all be renamed as well? (transitiveReferrers -> transitiveReferencers)

Reply:
Yes.  As far as my English anyway, `referrers` and `referencers` are synonyms, so we should pick one and use it everywhere.

Reply:
I actually changed them all back to referrers - apparently referencers is not a real word 😂